AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -66,9 +73,11 @@ - (BOOL)upsertWithToken:(MSTokenResult *)token
         documentWrapper:(MSDocumentWrapper *)documentWrapper
               operation:(NSString *_Nullable)operation
                 options:(MSBaseOptions *)options {
-  // Compute expiration time as now + device time to live (in seconds).
-  // If device time to live is set to infinite, set expiration time as null in the database.
-  // Note: If the cache/store is meant to be disabled, this method should not even be called.
+  /*
+   * Compute expiration time as now + device time to live (in seconds).
+   * If device time to live is set to infinite, set expiration time as null in the database.
+   * Note: If the cache/store is meant to be disabled, this method should not even be called.
+   */
   NSTimeInterval now = NSDate.timeIntervalSinceReferenceDate + NSTimeIntervalSince1970;
```

Comment:
Question: why not `[[NSDate date] timeIntervalSince1970]`? It would look easier to understand as for me.

Copy of [this comment](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1501#discussion_r274857858)

Reply:
Same question, why not use `timeIntervalSince1970` which is the pattern used everywhere in SDK.

Reply:
It is the same thing but it involves an additional and unnecessary memory allocation, so we did it this way

Reply:
Usually when we use `timeIntervalSince1970` we don't need to create a new `NSDate` object, but in this case we would.